### ISSUE TITLE
feature: Support for custom golang types as input values

### ISSUE THREAD

**Issue by komish** [windmill-labs/windmill#697]
I may be thinking about this the wrong way, which is why I figure this is a feature request and not a bug. To support nested objects in the data received from the user, I'm thinking the inner main function needs to support custom types, and overall that seems like it works from a UI perspective, but I end up with an undefined error.

Sample script, hacked together so it may also have some basic errors (forgive me for those)

```go
import (
    "fmt"
    "time"
)

type UserRequest struct {
    One string `json:"one"`    
    Two string `json:"two"`
    Three string `json:"three"`
}

type UserResponse struct {
    UserRequest `json:"req"`
    Received time.Time `json:"received"`
}

func main(req UserRequest) (interface{}, error) {
    fmt.Println("User Request", req)
    t := time.Now()
    return UserResponse{req, t}, nil
}
```

When I update the param from the default template to `req` and define it as a `UserRequest` type, the UI updates and lets me put in JSON without issue.

When hitting "Test", however, I see this error:

```
--- ERROR ---

Error during execution of the script
last 10 logs lines:
--- GO DEPENDENCIES SETUP---

go: creating new go.mod: module mymod
go: to add module requirements and sums:
	go mod tidy

--- GO CODE EXECUTION ---

# command-line-arguments
./main.go:25:51: undefined: UserRequest
```

I suppose potentially using a `map[string]interface{}` _might_ work, but I haven't gone into trying that out. 

The test data in the editor looks like this (It's a bit contrived):

```json
{
    "one": "1",
    "two": "2",
    "three": "3"
}
```

I think that's correct, and what I would pass using the webhook would probably look like this (barring typos):

```json
{
  "req": {
    "one": "1",
    "two": "2",
    "three": "3"
  }
}
```

In other words, achieving that nested object is the goal I was trying to test.
